frontend/src/views/main/project/Models.vue

```diff
@@ -36,55 +47,100 @@
                 </template>
                 <span>Download</span>
               </v-tooltip>
-              <DeleteModal v-if=" isProjectOwnerOrAdmin " :id=" m.id " :file-name=" m.fileName " type="model" @submit=" deleteModelPickle(m.id) " />
+              <DeleteModal v-if="isProjectOwnerOrAdmin" :id="m.id" :file-name="m.fileName" type="model" @submit=" deleteModelPickle(m.id)" />
             </div>
           </v-col>
         </v-row>
         <v-divider></v-divider>
       </v-card>
 
       <!-- Dialog for launching model inspection -->
-      <v-dialog persistent max-width="600" v-model=" showInspectDialog " class="inspector-launcher-container">
-        <InspectorLauncher :projectId=" projectId " :model=" modelToInspect " @cancel=" cancelLaunchInspector() " />
+      <v-dialog persistent max-width="600" v-model="showInspectDialog" class="inspector-launcher-container">
+        <InspectorLauncher :projectId="projectId" :model="modelToInspect" @cancel=" cancelLaunchInspector()" />
       </v-dialog>
 
     </v-container>
-    <v-container v-else class="font-weight-light font-italic secondary--text">
-      No models uploaded yet.
+    <v-container v-else>
+      <p class="font-weight-medium secondary--text">There are no models in this project yet. Follow the code snippet below to upload a model 👇</p>
+      <CodeSnippet :code-content="codeContent" :language="'python'"></CodeSnippet>
+      <p class="mt-4 font-weight-medium secondary--text">Check out the <a href="https://docs.giskard.ai/start/~/changes/QkDrbY9gX75RDMmAWKjX/guides/upload-your-model#2.-create-a-giskard-model" target="_blank">full documentation</a> for more information.</p>
     </v-container>
   </div>
 </template>
 
 <script setup lang="ts">
 import { api } from '@/api';
+import { apiURL } from "@/env";
+import { Role } from "@/enums";
 import InspectorLauncher from './InspectorLauncher.vue';
 import { ModelDTO } from '@/generated-sources';
 import mixpanel from "mixpanel-browser";
-import { onBeforeMount, ref } from 'vue';
+import { onBeforeMount, ref, computed } from 'vue';
 import DeleteModal from '@/views/main/project/modals/DeleteModal.vue';
 import InlineEditText from '@/components/InlineEditText.vue';
+import { useUserStore } from "@/stores/user";
+import { useProjectStore } from "@/stores/project";
 import { useMainStore } from "@/stores/main";
 import { useProjectArtifactsStore } from "@/stores/project-artifacts";
+import CodeSnippet from '@/components/CodeSnippet.vue';
 
+const userStore = useUserStore();
+const projectStore = useProjectStore();
 const projectArtifactsStore = useProjectArtifactsStore();
 
-const props = withDefaults(defineProps<{
+interface Props {
   projectId: number,
-  isProjectOwnerOrAdmin: boolean
-}>(), {
-  isProjectOwnerOrAdmin: false
-});
+}
+
+const props = defineProps<Props>();
 
 const showInspectDialog = ref<boolean>(false);
 const modelToInspect = ref<ModelDTO | null>(null);
 
+const codeContent = computed(() =>
```

Comment:
can you add a button (or some other way) to show this snippet even if there are existing models (same for datasets?) 
It's a pity to hide it forever, it may be useful to show on demand for new uploads

Reply:
> can you add a button (or some other way) to show this snippet even if there are existing models (same for datasets?) It's a pity to hide it forever, it may be useful to show on demand for new uploads

I considered linking this "Upload with API" button on the top right to open the quick start guide modal on the upload artifacts step.
![image](https://github.com/Giskard-AI/giskard/assets/44180294/94018ad8-65db-4a1a-9b5e-bf0457d682ec)

What do you think?

Reply:
yes, good idea. You probably don't need to show the whole guide when you click on the upload with API, just this snipped again and a link to the docs

Reply:
@henchaves , I thought we wanted to open a modal with a code snipped, didn't we?

https://github.com/Giskard-AI/giskard/assets/7889985/fc2cfeb0-3d2e-4a91-bd55-0e8e24e72281

Reply:
@andreybavt I'm doing that in another branch [https://linear.app/giskard/issue/GSK-1061/catalog-tab-ui-improvements](https://linear.app/giskard/issue/GSK-1061/catalog-tab-ui-improvements), because other code snippets will be added in this one (usage examples of each test)